File: DarkOrbit 7.5.3/Net/netty/commands/ClientUITextReplacementModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class ClientUITextReplacementModule
    {
        public static short ID = 1059;

        public String mReplacement = "";
        public String mWildcard = "";
        public ClientUITooltipTextFormatModule mTooltipTextFormat;

        public ClientUITextReplacementModule(String pWildCard, ClientUITooltipTextFormatModule pTooltipTextFormat,
                                         String pReplacement)
        {
            this.mWildcard = pWildCard;
            this.mReplacement = pReplacement;
            this.mTooltipTextFormat = pTooltipTextFormat;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeUTF(this.mReplacement);
            param1.write(mTooltipTextFormat.write());
            param1.writeShort(28496);
            param1.writeUTF(this.mWildcard);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/ClientUITextReplacementModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

public class ClientUITextReplacementModule
        implements ServerCommand {

    public static int ID = 1020;

    public String mReplacement = "";

    public String mWildcard = "";

    public ClientUITooltipTextFormatModule mTooltipTextFormat;

    public ClientUITextReplacementModule(String pWildCard, ClientUITooltipTextFormatModule pTooltipTextFormat,
                                         String pReplacement) {
        this.mWildcard = pWildCard;
        this.mReplacement = pReplacement;
        this.mTooltipTextFormat = pTooltipTextFormat;
    }

    public int getID() {
        return ID;
    }

    public int method_1005() {
        return 4;
    }

    public void write(DataOutputStream out) {
        try {
            out.writeShort(ID);
            this.writeInternal(out);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream out) {
        try {
            this.mTooltipTextFormat.write(out);
            out.writeUTF(this.mWildcard);
            out.writeUTF(this.mReplacement);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}